AppDB/restore_data.py

```diff
@@ -0,0 +1,301 @@
+""" This process performs a restore of all the application entities from a
+given restore.
+"""
+import argparse
+import cPickle
+import glob
+import logging
+import multiprocessing
+import os
+import random
+import sys
+import time
+
+import appscale_datastore_batch
+from backup_data import DatastoreBackup
+import datastore_server
+import delete_all_records
+
+from google.appengine.api import datastore_errors
+from google.appengine.datastore import datastore_pb
+from google.appengine.datastore import entity_pb
+
+from zkappscale import zktransaction as zk
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../lib/"))
+import appscale_info
+
+_APPS_LOCATION = '/var/apps/'
+
+class DatastoreRestore(multiprocessing.Process):
+  """ Backs up all the entities for a set application ID. """
+
+  # The amount of time to wait for SIGINT in seconds.
+  SIGINT_TIMEOUT = 5
+
+  # The number of entities retrieved in a datastore request.
+  BATCH_SIZE = 100
+
+  # Retry sleep on datastore error in seconds.
+  DB_ERROR_PERIOD = 30
+
+  # The amount of seconds between polling to get the restore lock.
+  LOCK_POLL_PERIOD = 60
+
+  def __init__(self, app_id, backup_dir, zoo_keeper, table_name):
+    """ Constructor.
+
+    Args:
+      app_id: A str, the application ID.
+      backup_file: A str, the location of the backup file.
+      zoo_keeper: A ZooKeeper client.
+      table_name: The database used (e.g. cassandra).
+    """
+    multiprocessing.Process.__init__(self)
+
+    self.app_id = app_id
+    self.backup_dir = backup_dir
+    self.zoo_keeper = zoo_keeper
+    self.table = table_name
+
+    self.entities_restored = 0
+    self.indexes = []
+    self.ds_distributed = None
+
+  def stop(self):
+    """ Stops the restore process. """
+    pass
+
+  def run(self):
+    """ Starts the main loop of the restore thread. """
+    while True:
+      logging.debug("Trying to get restore lock.")
+      if self.get_restore_lock():
+        logging.info("Got the restore lock.")
+
+        datastore_batch = appscale_datastore_batch.\
```

Comment:
Should these two creations of constructors be outside the while loop rather than be created each time?